misc/scripts/models-as-data/generate_flow_model.py

```diff
@@ -141,58 +127,57 @@ def runQuery(self, query):
 
 
     def asAddsTo(self, rows, predicate):
-        if rows.strip() == "":
-            return ""
-        return helpers.addsToTemplate.format(f"codeql/{self.language}-all", predicate, rows)
-
+        extensions = { }
+        for key in rows:
+            extensions[key] = helpers.addsToTemplate.format(f"codeql/{self.language}-all", predicate, rows[key])
+        return extensions
 
     def getAddsTo(self, query, predicate):
         data = self.runQuery(query)
         rows = parseData(data)
         return self.asAddsTo(rows, predicate)
 
-
     def makeContent(self):
         if self.generateSummaries:
             summaryAddsTo = self.getAddsTo("CaptureSummaryModels.ql", helpers.summaryModelPredicate)
         else:
-            summaryAddsTo = ""
+            summaryAddsTo = { }
 
         if self.generateSinks:
             sinkAddsTo = self.getAddsTo("CaptureSinkModels.ql", helpers.sinkModelPredicate)
         else:
-            sinkAddsTo = ""
+            sinkAddsTo = { }
 
         if self.generateSources:
             sourceAddsTo = self.getAddsTo("CaptureSourceModels.ql", helpers.sourceModelPredicate)
         else:
-            sourceAddsTo = ""
+            sourceAddsTo = {}
 
         if self.generateNeutrals:
             neutralAddsTo = self.getAddsTo("CaptureNeutralModels.ql", helpers.neutralModelPredicate)
         else:
-            neutralAddsTo = ""
+            neutralAddsTo = { }
         
-        return f"""# THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
-# Definitions of models for the {self.friendlyname} framework.
-extensions:
-{sinkAddsTo}{sourceAddsTo}{summaryAddsTo}{neutralAddsTo}"""
+        return helpers.merge(summaryAddsTo, sinkAddsTo, sourceAddsTo, neutralAddsTo)
 
     def makeTypeBasedContent(self):
         if self.generateTypeBasedSummaries:
-            typeBasedSummaryAddsTo = self.getAddsTo("CaptureTypeBasedSummaryModels.ql", "extSummaryModel")
+            typeBasedSummaryAddsTo = self.getAddsTo("CaptureTypeBasedSummaryModels.ql", helpers.summaryModelPredicate)
         else:
-            typeBasedSummaryAddsTo = ""
+            typeBasedSummaryAddsTo = { }
 
-        return f"""# THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
-# Definitions of type based summaries in the {self.friendlyname} framework.
-extensions:
-{typeBasedSummaryAddsTo}"""
+        return typeBasedSummaryAddsTo
 
-    def save(self, content, target):
-        with open(target, "w") as targetYml:
-            targetYml.write(content)
-        print("Models as data extensions written to " + target)
+    def save(self, extensions, extension):
+        # Create a file for each namespace and save models.
+        extensionTemplate = """# THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
+extensions:
+{0}"""
+        for entry in extensions:
+            target = os.path.join(self.generatedFrameworks, entry + extension)
```

Comment:
Did you also try running this for java?

Reply:
Nope :-)

Reply:
If you are interested then you can run modify the `mad_modelDiff.yml` workflow file to also be triggered on the shared python code.

Reply:
I don't know how to run that on my PR branch, do you?

Reply:
Can't you just change the paths section of the workflow file? If that doesn't work - then I don't know.